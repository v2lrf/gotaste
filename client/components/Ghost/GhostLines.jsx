import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import GhostCard from './GhostCard'

function GhostLines({ lines }) {
  return Array(lines)
    .fill()
    .map((v, k) => k)
    .map(val => (
      <Fragment key={val}>
        <div className="mb-2">
          <GhostCard height={16} width={100} />
        </div>
        <div className="mb-2">
          <GhostCard height={16} width={75} />
        </div>
      </Fragment>
    ))
}

GhostLines.propTypes = {
  lines: PropTypes.number
}

GhostLines.defaultProps = {
  lines: 2
}

export default GhostLines
