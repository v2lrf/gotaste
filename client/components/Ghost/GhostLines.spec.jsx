import React from 'react'

import GhostLines from './GhostLines'

describe('GhostLines', () => {
  let component
  beforeEach(() => {
    component = shallow(<GhostLines />)
  })

  describe('Snapshots', () => {
    it('should match the default snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
