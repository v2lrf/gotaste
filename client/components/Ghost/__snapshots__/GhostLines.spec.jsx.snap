// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GhostLines Snapshots should match the default snapshot 1`] = `
Array [
  <Fragment
    key="0"
  >
    <div
      className="mb-2"
    >
      <GhostCard
        height={16}
        width={100}
      />
    </div>
    <div
      className="mb-2"
    >
      <GhostCard
        height={16}
        width={75}
      />
    </div>
  </Fragment>,
  <Fragment
    key="1"
  >
    <div
      className="mb-2"
    >
      <GhostCard
        height={16}
        width={100}
      />
    </div>
    <div
      className="mb-2"
    >
      <GhostCard
        height={16}
        width={75}
      />
    </div>
  </Fragment>,
]
`;
